File: workspace/nodejs-basic/04_server/exam02.js
// 모듈을 가져온다
var http = require("http");

// 서버 생성 (누군가 요청이 있을때 서버 생성 됨)
// createServer(function - 서버 요청시에 자동 호출되는 함수)
var server = http.createServer(function(request, response) {
	//response.writeHead(statusCode: 응답코드, headers)
	response.writeHead(200, {"Content-Type": "text/html; charset=UTF-8"});
	
	// 여러줄의 데이트를 보낼때는 write 사용
	response.write("<h1>화면에 데이터 보내기</h1>");
	response.write("<h1>노드를 이용한 출력</h1>");
	
	// 요청에 의한 응답을 보내줌 
	response.end("<h1>Hello!! 노드~~");
});

server.listen(10001, function() {
	console.log("10001번 포트로 서버 구동중...");
});

File: ServletJsp/src/servlet/quiz/Exam02.java
package servlet.quiz;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.Random;

import javax.servlet.GenericServlet;
import javax.servlet.ServletConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebServlet;

@WebServlet("/servlet/quiz/Exam02")
public class Exam02 extends GenericServlet {
	private int answerNum;
	private int count = 0;
	public void init(ServletConfig config) {
		Random r = new Random();
		answerNum = r.nextInt(100) + 1;
	}
	@Override
	public void service(ServletRequest request, ServletResponse response) throws ServletException, IOException {
		response.setContentType("text/html; charset=UTF-8");
		PrintWriter out = response.getWriter();
		
		String msg = "";
		int num = Integer.parseInt(request.getParameter("num"));
		if (num > answerNum) {
			msg = "작은 수를 입력하세요";
		}
		else if (num < answerNum) {
			msg = "큰 수를 입력하세요";
		}
		else {
			msg = "정답입니다.";
		}
		count++;
		
		out.println("<html><body>");
		out.println("  <h1>정답 : " + answerNum + "</h1>");
		out.println("  <h1>" + count + "회 시도</h1>");
		out.println("  <h1>" + msg + "</h1>");
		if (num != answerNum) {
			out.println("  <h1><a href='exam02.html'>다시입력</a></h1>");
		}
		out.println("</body></html>");
	}
}





